src/dbnode/storage/index/convert/convert.go

```diff
@@ -191,7 +191,7 @@ func FromSeriesIDAndTagIter(id ident.ID, tags ident.TagIterator) (doc.Metadata,
 }
 
 // FromSeriesIDAndEncodedTags converts the provided series id and encoded tags into a doc.Metadata.
-func FromSeriesIDAndEncodedTags(id ident.BytesID, encodedTags ts.EncodedTags) (doc.Metadata, error) {
+func FromSeriesIDAndEncodedTags(id ident.ID, encodedTags ts.EncodedTags) (doc.Metadata, error) {
```

Comment:
Can we avoid this switch to an interface? Using `ident.BytesID` was very intentional here, to avoid allocations in peer bootstrapping and some other use cases: https://github.com/m3db/m3/pull/3087#discussion_r560125931